### PR TITLE
Add support to set color when creating a ScopedZone

### PR DESCRIPTION
**Pull request by Chekov2k**
This pull request adds the option to specify the colour of a scoped zone for the c-api as well as for the coming Python API.

As requested, this has been split out from [Pull Request 754](https://github.com/wolfpld/tracy/pull/754).

I hope adding a default colour parameter to `tracy::Profiler::AllocSourceLocation` is okay as well as using constructor delegation for the `ScopedZone`.

### CODE REVIEW

**Comment by wolfpld** [comment]
Where are these new constructors used or how are they exposed?

**Comment by Chekov2k** [comment]
> Where are these new constructors used or how are they exposed?

Apologies, I just realised that I apparently did not force push to update my Pull-Request correctly :-( I also added some more explanation below.

The main use for me is in the `PyScopedZone` used to create the Python-bindings:

```cpp
fZone = new tracy::ScopedZone(fLine, fFile.c_str(), fFunction.c_str(), fName.c_str(), fName.size(), fColor, true);
```

The other changes are to expose them in the C-API as requested by [simonvanbernem](https://github.com/simonvanbernem) in [Pull-Request 754](https://github.com/wolfpld/tracy/pull/754), i.e. in `___tracy_alloc_srcloc` and `___tracy_alloc_srcloc_name`

**Comment by Chekov2k** [comment]
Had to add a static_cast<uint32_t>(0) so the ScopedZone constructor is delegating to the right one ...

File: public/tracy/TracyC.h [hunk 1]
```diff
@@ -182,8 +182,8 @@ TRACY_API void ___tracy_startup_profiler(void);
 TRACY_API void ___tracy_shutdown_profiler(void);
 #endif
 
-TRACY_API uint64_t ___tracy_alloc_srcloc( uint32_t line, const char* source, size_t sourceSz, const char* function, size_t functionSz );
-TRACY_API uint64_t ___tracy_alloc_srcloc_name( uint32_t line, const char* source, size_t sourceSz, const char* function, size_t functionSz, const char* name, size_t nameSz );
+TRACY_API uint64_t ___tracy_alloc_srcloc( uint32_t line, const char* source, size_t sourceSz, const char* function, size_t functionSz, uint32_t color = 0 );
+TRACY_API uint64_t ___tracy_alloc_srcloc_name( uint32_t line, const char* source, size_t sourceSz, const char* function, size_t functionSz, const char* name, size_t nameSz, uint32_t color = 0 );
```
[the remark this answers is not in the document]
Comment (wolfpld) [reply public/tracy/TracyC.h:186]: No, keep the current one. It's an easy to fix API break.

[the remark this answers is not in the document]
Comment (Chekov2k) [reply public/tracy/TracyC.h:186]: Will do!

**Pull request merged**